### PR TITLE
handler tests: ensure events emiited

### LINKED ISSUE #664: handler tests: ensure events emitted
In the handler tests, ensure that the correct events are emitted.  #662 implements this for `x/deployment/handler`, the other two modules still need to be tested.

* [ ] `x/market/handler`
* [ ] `x/provider/handler`

### PR DESCRIPTION
**Pull request by akhilkumarpilli**
Fixes #664

### CODE REVIEW

File: x/deployment/types/events_test.go [hunk 1]
```diff
@@ -25,15 +26,12 @@ type testEventParsing struct {
...
-				t.Errorf("unexpected error: %v exp: %v", err, tep.expErr)
-				t.Logf("%T %v", errors.Cause(err), err)
-				t.Logf("%+v", tep)
-			}
+		t.Logf("%+v", tep)
+		// if the error expected is errWildcard to catch untyped errors, don't fail the test, the error was expected.
+		if errors.Is(tep.expErr, errWildcard) {
+			require.Error(t, err)
+		} else {
+			require.Equal(t, tep.expErr, err)
```
Comment (boz) [inline x/deployment/types/events_test.go:34]: thanks.

File: x/market/handler/handler_test.go [hunk 2]
```diff
@@ -97,7 +97,18 @@ func TestCreateBidValid(t *testing.T) {
...
+	bid := types.MakeBidID(order.ID(), provider)
+
+	t.Run("ensure event created", func(t *testing.T) {
+		iev := testutil.ParseMarketEvent(t, res.Events[2:])
+		require.IsType(t, types.EventBidCreated{}, iev)
+
+		dev := iev.(types.EventBidCreated)
+
+		require.Equal(t, bid, dev.ID)
+	})
```
Comment (boz) [inline x/market/handler/handler_test.go:109]: :raised_hands:

File: x/market/handler/handler_test.go [hunk 3]
```diff
@@ -313,6 +333,15 @@ func TestCloseBidValid(t *testing.T) {
...
 	require.NotNil(t, res)
 	require.NoError(t, err)
+
+	t.Run("ensure event created", func(t *testing.T) {
+		iev := testutil.ParseMarketEvent(t, res.Events[3:4])
+		require.IsType(t, types.EventBidClosed{}, iev)
+
+		dev := iev.(types.EventBidClosed)
+
+		require.Equal(t, msg.BidID, dev.ID)
```
Comment (boz) [inline x/market/handler/handler_test.go:343]: I don't think it's needed here.  If anything, we want to make sure that `EventCloseBid` is _not_ emitted.

File: x/market/types/event.go [hunk 4]
```diff
@@ -18,6 +18,9 @@ const (
 	evActionLeaseCreated = "lease-created"
 	evActionLeaseClosed  = "lease-closed"
 
+	evOwnerKey       = "owner"
+	evDSeqKey        = "dseq"
+	evGSeqKey        = "gseq"
```
Comment (boz) [inline x/market/types/event.go:23]: can you move these to the test file?  I think it's the only place that they're used.

File: x/market/types/event.go [hunk 5]
```diff
@@ -223,7 +226,7 @@ func parseEVPriceAttributes(attrs []sdk.Attribute) (sdk.Coin, error) {
 }
 
 // ParseEvent parses event and returns details of event and error if occurred
-func ParseEvent(ev sdkutil.Event) (interface{}, error) {
+func ParseEvent(ev sdkutil.Event) (sdkutil.ModuleEvent, error) {
```
Comment (boz) [inline x/market/types/event.go:229]: :+1:

File: x/provider/types/event.go [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+	evActionProviderCreate = "provider-create"
+	evActionProviderUpdate = "provider-update"
+	evActionProviderDelete = "provider-delete"
+	evOwnerKey             = "owner"
+)
+
+// EventProviderCreate struct
+type EventProviderCreate struct {
+	Owner sdk.AccAddress
+}
```
Comment (boz) [inline x/provider/types/event.go:18]: Can you add the events from this file to a new section [here](https://github.com/ovrclk/akash/blob/189a668f583e03adc47a1c0ecb70fd0ca238debf/events/publish_test.go#L16-L19)?

I think that you'll need to update [`processEvent`](https://github.com/ovrclk/akash/blob/189a668f583e03adc47a1c0ecb70fd0ca238debf/events/publish.go#L98-L100) too.

File: x/market/handler/handler_test.go (hunk 3 above)
Comment (akhilkumarpilli) [reply x/market/handler/handler_test.go:343]: Sorry, I didn't get you. This is to ensure `EventBidClosed` emitted when valid bid is closed.

Comment (boz) [reply x/market/handler/handler_test.go:343]: Sorry, the diff made it look like this was in `TestCloseBidNonExisting`.  You're right - it belongs here.

**Pull request merged**